### ISSUE TITLE
Support for Markdown files

### ISSUE THREAD

**Issue by tdudgeon** [InformaticsMatters/mini-apps-data-tier-ui#683]
We need to support the ability for users to document their work in projects.
A simple way of handling Markdown files, and special treatment of README.md in a way similar to how GitHub does this would go a long way towards this.

Need to be able to:
- display formatted Markdown files in a viewer
- if a README.md file is present then display it's contents as part of the directory listing
- allow to edit Markdown files
- allow to incorporate links to other files in the project (and external URLs) into the Markdown content

**Comment by OliverDudgeon** [comment]
Display of markdown should be quite easy since we already display mdx files
Editing is more difficult but is possible via Jupyter currently

**Comment by tdudgeon** [comment]
Yes, editing can be done in Jupyter for now, but I bet there's a library that we can use that allows this.

**Comment by OliverDudgeon** [comment]
Yes, the engine behind VSCode can be deployed which supports a lot of stuff including markdown.
